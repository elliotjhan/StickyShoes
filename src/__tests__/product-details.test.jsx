import React from "react";
import ProductDetails from "./../components/product-details";
import { render, screen, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";

afterEach(() => {
  cleanup();
})

const numberWithCommas = (number) => {
  let newNumber = (parseFloat(number)).toFixed(2);
  return newNumber.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
}

test("Product details should have 3 pictures, name, price, and description", () => {
  const currentProduct = {
    image: 'shaman1.jpeg',
    name: 'Evolv Shaman',
    price: 165,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.',
    carousel: ['shaman1.jpeg', 'shaman2.jpeg', 'shaman3.jpeg'],
    productid: 2
  }
  render(
    <ProductDetails
      numberWithCommas={numberWithCommas}
      currentProduct={currentProduct}
      productList={[]}
    />, { wrapper: MemoryRouter }
  );
  const productDetails = screen.getByTestId("productDetails");
  expect(productDetails).toHaveTextContent(currentProduct.name);
  expect(productDetails).toHaveTextContent(currentProduct.price);
  expect(productDetails).toHaveTextContent(currentProduct.description);
});

test('matches snapshot', () => {
  const currentProduct = {
    image: 'shaman1.jpeg',
    name: 'Evolv Shaman',
    price: 165,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.',
    carousel: ['shaman1.jpeg', 'shaman2.jpeg', 'shaman3.jpeg'],
    productid: 2
  }
  let tree = create(
    <MemoryRouter>
      <ProductDetails
        numberWithCommas={numberWithCommas}
        currentProduct={currentProduct}
        productList={[]}
      />
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})