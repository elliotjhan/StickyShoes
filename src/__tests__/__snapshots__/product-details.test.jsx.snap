// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="row align-items-center justify-content-center productDetailsContainer"
  data-testid="productDetails"
>
  <div
    className="col-6"
  />
  <div
    className="col-6"
  >
    <div
      className="productDetailsName"
    />
    <br />
    <div
      className="productDetailsPrice"
    >
      $
    </div>
    <br />
    <div
      className="productDetailsDescription"
    />
    <br />
    <div
      className="row align-items-center text-start"
    >
      <div
        className="col-2"
      >
        <div
          className="quantity-input"
        >
          <button
            className="quantity-input__modifier quantity-input__modifier--left"
            onClick={[Function]}
          >
            —
          </button>
          <input
            className="quantity-input__screen"
            readOnly={true}
            type="text"
            value={1}
          />
          <button
            className="quantity-input__modifier quantity-input__modifier--right"
            onClick={[Function]}
          >
            ＋
          </button>
        </div>
      </div>
      <div
        className="col-3"
      >
        <button
          className="cartButton"
          onClick={[Function]}
        >
          Add To Cart
        </button>
      </div>
    </div>
    <div
      className="row keepShoppingContainer"
    >
      <div
        className="col"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "textDecoration": "none",
            }
          }
        >
          <span
            className="cursor keepShopping"
          >
            &lt;Keep Shopping
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
